directories/src/redux/epics/node-epics.js

```diff
@@ -143,9 +147,34 @@ const requestGenesisHashEpic = (action$, store) => {
             brokerNode.completeGenesisHashPoW(txid, trytesArray[0])
           ).map(({ data }) => {
             const { purchase: genesisHash, numberOfChunks } = data;
-            return nodeActions.addGenesisHash({ genesisHash, numberOfChunks });
-          })
+            return { genesisHash, numberOfChunks };
+          }).catch(error => {
+            console.log("GENESIS HASH COMPLETE ERROR", error);
+            return Observable.empty();
+          });
         )
+        .mergeMap(({ genesisHash, numberOfChunks }) => {
+          const datamap = Datamap.generate(genesisHash, numberOfChunks);
+          const addresses = _.values(datamap);
+          const randomNumber = AppUtils.randomNumber(addresses.length);
+          const randomSectorAddresses = addresses.slice(randomNumber);
```

Comment:
What will randomSectorAddresses look like?  It needs to be something like 0 to 999,999 (if the random sector selected was sector 1 or 1,000,000 to 1,999,999 (if the random sector selected was sector 2)

Reply:
It will randomly select one address

Reply:
So what you are saying is that you will pick a random address.  Then you will determine what sector that address is in.  Then, that sector will be your randomly chosen sector.  

I.e. say your random address was 1,250,000.  This would mean that you'd be looking at addresses 1,000,0000 through 1,999,999.  Is that how you are approaching this?